src/components/NewRoomModal.tsx

```diff
@@ -0,0 +1,167 @@
+'use client'
+import { useState, useRef, useCallback, useEffect } from 'react'
+import { useRouter } from 'next/navigation'
+
+// import { socket } from '@/socket'
+
+import RightArrowIcon from './icons/RightArrowIcon'
+
+export default function NewRoomModal() {
+	const [isLoading, setIsLoading] = useState(true)
+	const [roomUrl, setRoomUrl] = useState('')
+	const [hostRoomUrl, setHostRoomUrl] = useState('')
+	const [hideGoBtn, setHideGoBtn] = useState(true)
+	const [hideCreateBtn, setHideCreateBtn] = useState(true)
+	const [showNameInput, setShowNameInput] = useState(true)
+	const [domain, setDomain] = useState('')
+	const roomUrlRef = useRef<HTMLSpanElement>(null)
+	const dialogRef = useRef<HTMLDialogElement>(null)
+	const hostRef = useRef<HTMLInputElement>(null)
+	const [hostFirstName, setHostFirstName] = useState('')
+
+	const router = useRouter()
+
+	useEffect(() => {
+		if (typeof window !== 'undefined') {
```

Comment:
question/todo: Why did you check if window is undefined here? In useEffect, it always exists because this hook only runs on the client/browser.

Reply:
I will eliminate that. I only did that because that's the example I saw and then put it in the useEffect.